module3/homework/advanced_calculator/advancedCalculator.cpp

```diff
@@ -0,0 +1,114 @@
+#include "advancedCalculator.hpp"
+
+#include <algorithm>
+#include <array>
+#include <cmath>
+#include <functional>
+#include <map>
+#include <regex>
+#include <string_view>
+
+constexpr std::string_view ERROR_CODE_MESSAGE[] = {"Result: ",
+                                                   "Bad Format!",
+                                                   "You cant divide by zero!",
+                                                   "Bad Character!",
+                                                   "You cant get sqrt from negative number!",
+                                                   "You cant get module from non integer value!"};
+
+constexpr size_t ALLOWED_CHARACKTERS_SIZE = 11;
+constexpr std::array<char, ALLOWED_CHARACKTERS_SIZE> ALLOWED_CHARACTERS = {'+', '-', '*', '/', '%', '^', '$', '!', '.', ' ', ','};
+
+const std::map<std::string, std::function<double(double, double)>> calculateFunctions{
+    {"+", std::plus<double>()},
+    {"-", std::minus<double>()},
+    {"*", std::multiplies<double>()},
+    {"/", std::divides<double>()},
+    {"%", std::modulus<int>()},
+    {"^", [](const double base, const double exponent) { return pow(base, exponent); }},
+    {"$", [](const double number, const double root) { return pow(number, 1 / root); }},
+    {"!", [](const double value, const double defaultValue) { return (value < 0) ? defaultValue : tgamma(value + defaultValue); }}};
```

Comment:
Please make separate function factorial, and move code there. Will be easier to read :)